lib/sourcemapper.js

```diff
@@ -0,0 +1,173 @@
+ *  processing the given sourcemap files
+ * @constructor
+ */
+function Sourcemapper(sourcemapPaths, logger) {
+  this.infoMap = new Map();
+
+  var that = this;
+  _.forEach(sourcemapPaths, function(fullMapPath) {
+    // this handles the case when the path is undefined, null, or
+    // the empty string
+    if (!fullMapPath || !polyfill.endsWith(fullMapPath, MAP_EXT)){
+      return;
+    }
+    fullMapPath = path.normalize(fullMapPath);
+
+    fs.readFile(fullMapPath, 'utf8',
+      function(err, data) {
+        if (err){
+          logger.error('Could not read sourcemap file ' + fullMapPath +
+                       ': ' + err);
+          return;
+        }
+
+        try {
+          var consumer = new sourceMap.SourceMapConsumer(data);
+
+          /*
+           * If the sourcemap file defines a "file" attribute, use it as
+           * the output file where the path is relative to the directory
+           * containing the map file.  Otherwise, use the name of the output
+           * file (with the .map extension removed) as the output file.
+           */
+          var outputBase = consumer.file ? consumer.file
+                                         : path.basename(fullMapPath, '.map'),
+              parentDir = path.dirname(fullMapPath),
+              outputPath = path.join(parentDir, outputBase);
+          
+          var sources = Array.prototype.slice.call(consumer.sources)
+            .filter(function(value) {
+              // filter out any empty string, null, or undefined sources
+              return !!value;
+            })
+            .map(function(relPath) {
+              // resolve the relative paths to absolute paths
+              return path.join(parentDir, relPath);
+            });
+
+          if (sources.length === 0) {
+            logger.error('No sources listed in the sourcemap file ' +
+                         fullMapPath);
+            return;
+          }
+
+          _.forEach(sources, function(src) {
+            that.infoMap.set(path.normalize(src), {
+              outputFile: outputPath,
+              mapFile: fullMapPath,
+              mapConsumer: consumer
+            });
+          });
+        }
+        catch(e) {
+          logger.error('An error occurred while reading the '+
+                       'sourcemap file ' + fullMapPath + ': ' + err);
+        }
+    });
+  });
+}
+
+/**
+ * Used to determine if the source file specified by the given path has
+ * a .map file and an output file associated with it.
+ * 
+ * If there is no such mapping, it could be because the input file is not
+ * the input to a transpilation process or it is the input to a transpilation
+ * process but its corresponding .map file was not given to the constructor
+ * of this mapper.
+ * 
+ * @param {string} inputPath The absolute path to an input file that could
+ *  possibly be the input to a transpilation process
+ */
+Sourcemapper.prototype.hasMappingInfo = function(inputPath) {
```

Comment:
The `Sourcemapper` has been updated so that the paths provided to its constructor are paths relative to the process's current working directory instead of absolute paths.  This seems to be more general than using the project's working directory, and it seems that the `node` process needs to be run from within the project's directory so that the `package.json` file can be located.  Let me know if this is not the case and/or you see any problems with this approach.